### ISSUE TITLE
False-positive E1121 with self in subclass of np.ndarray

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#2476]
### Steps to reproduce

```python
import numpy as np


class MyArray(np.ndarray):
    def __new__(cls, arr):
        arr = np.asanyarray(arr).view(cls)
        return arr

    def __getitem__(self, idx):
        sub = self.view(np.ndarray).__getitem__(idx)
        return sub


print(MyArray([5, 6, 7]))
```

### Current behavior
Test.py:10:14: E1121: Too many positional arguments for method call (too-many-function-args)

### Expected behavior
No error

### pylint --version output
pylint 2.1.1
astroid 2.0.4
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 17:00:18) [MSC v.1900 64 bit (AMD64)]

**Comment by mbyrnepr2** [comment]
I can't reproduce using the newest Pylint:
```python
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.10.4
```

Will close but please reopen if any queries.
